mail_tracking/static/src/js/failed_message.js

```diff
@@ -0,0 +1,366 @@
+/* Copyright 2019 Alexandre Díaz
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). */
+odoo.define('mail_tracking.FailedMessage', function (require) {
+    "use strict";
+
+    var ChatAction = require('mail.chat_client_action');
+    var AbstractField = require('web.AbstractField');
+    var BasicModel = require('web.BasicModel');
+    var BasicView = require('web.BasicView');
+    var Chatter = require('mail.Chatter');
+    var utils = require('mail.utils');
+    var chat_manager = require('mail.chat_manager');
+    var core = require('web.core');
+    var field_registry = require('web.field_registry');
+    var time = require('web.time');
+    var session = require('web.session');
+    var config = require('web.config');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+
+    /* DISCUSS */
+    var failed_counter = 0;
+    var is_channel_failed_outdated = false;
```

Comment:
Based on the usage I see here, you'll possibly better use a deferred.

```suggestion
    var is_channel_failed_outdated = $.Deferred();
```

Reply:
I can't see the benefits of use deferred... The main problem is that can't modify "channels": https://github.com/odoo/odoo/blob/11.0/addons/mail/static/src/js/chat_manager.js#L385

Reply:
Hmm OK, maybe I misunderstood something.

We can handle that in the migration if needed, I think in v12 this will be easier to write.